lsp-methods.el

```diff
@@ -751,9 +748,14 @@ interface TextDocumentEdit {
         (lsp--apply-text-edits (gethash "edits" edit))))))
 
 (defun lsp--text-edit-sort-predicate (e1 e2)
```

Comment:
Comparing both end point and start point addresses one corner case: when one range is empty and the other shares the same end point with it, i.e.  [3, 3) and [1, 3).  We want [3, 3) come first.

Since text edits are guaranteed to be non-overlapping. This can be simplified to compare with their middle point, i.e.
`e1.start + e1.end > e2.start + e2.end` (division by 2 on both sides are omitted).

Reply:
Since 3 > 1, wouldn't `[3,3)` be sorted before `[1,3)` anyway?